3scaleAdapter/pkg/threescale/metrics/prometheus_test.go

```diff
@@ -0,0 +1,97 @@
+package metrics
+
+import (
+	"testing"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus/testutil"
+)
+
+func TestObserveSystemLatency(t *testing.T) {
+	const metricName = "threescale_system_latency"
+	const expect = `
+
+                # HELP threescale_system_latency Request latency for requests to 3scale system URL
+                # TYPE threescale_system_latency histogram
+                threescale_system_latency_bucket{serviceID="123",systemURL="www.fake.com",le="0.05"} 0
+                threescale_system_latency_bucket{serviceID="123",systemURL="www.fake.com",le="0.08"} 0
+                threescale_system_latency_bucket{serviceID="123",systemURL="www.fake.com",le="0.1"} 0
+                threescale_system_latency_bucket{serviceID="123",systemURL="www.fake.com",le="0.15"} 0
+                threescale_system_latency_bucket{serviceID="123",systemURL="www.fake.com",le="0.2"} 0
+                threescale_system_latency_bucket{serviceID="123",systemURL="www.fake.com",le="0.3"} 0
+                threescale_system_latency_bucket{serviceID="123",systemURL="www.fake.com",le="0.5"} 0
+                threescale_system_latency_bucket{serviceID="123",systemURL="www.fake.com",le="1"} 0
+                threescale_system_latency_bucket{serviceID="123",systemURL="www.fake.com",le="1.5"} 1
+                threescale_system_latency_bucket{serviceID="123",systemURL="www.fake.com",le="+Inf"} 1
+                threescale_system_latency_sum{serviceID="123",systemURL="www.fake.com"} 1.001
+                threescale_system_latency_count{serviceID="123",systemURL="www.fake.com"} 1
+
+        `
+	r := NewMetricsReporter(true, 8080)
+	r.ObserveSystemLatency("www.fake.com", "123", time.Second+time.Millisecond)
+	// TODO - Uncomment when https://github.com/prometheus/client_golang/issues/494 is resolved
```

Comment:
this issue is now closed - want to wait for the histogram equivalent and release?

Reply:
Yes, this bug fix didn't resolve it, I've submitted a fix for all other collector types. So it's still broken, but I can update the link and lets merge this anyway when ready. 

No point waiting for a release to be cut upstream IMO (I've deployed and visually tested expected behaviour here) and as I assume we don't want to point `dep` at a branch or commit?

Reply:
not worth it for these tests if we expect such release to occur soon

Reply:
Agree. Not sure when the release will happen but the fix will happen pretty quickly